plugins/modules/packaging/os/pkgng.py

```diff
@@ -335,7 +363,8 @@ def main():
     module = AnsibleModule(
         argument_spec=dict(
             state=dict(default="present", choices=["present", "latest", "absent"], required=False),
-            name=dict(aliases=["pkg"], required=True, type='list'),
+            name=dict(aliases=["pkg"], default=[], required=False, type='list'),
```

Comment:
Do I understand it correctly that `state` and `name` are ignored if `upgrade` is `True`? This needs to be properly documented.

Also, I would make `name` required if `upgrade` is not `True`.

Reply:
> Do I understand it correctly that `state` and `name` are ignored if `upgrade` is `True`? This needs to be properly documented.

No, `state` and `name` can be passed along with `upgrade`. The module will first perform the upgrade and then make state changes to packages.

> 
> Also, I would make `name` required if `upgrade` is not `True`.

Sure, I'll prepare a patch for that in the morning, leaving work now.

Reply:
I'm actually not sure how to achieve this with the provided checks in AnsibleModule. I'd need to make ``name`` required if both of ``upgrade`` or ``autoremove`` are false.

But after thinking it through, I don't think it's needed. The module can run with any combination of ``name``, ``upgrade`` and ``autoremove``, including none or all of them. If none are passed, the module will do nothing. What do you think?

Also, just wanted to say thanks for your help with this and the other PR that got merged yesterday. The PR process has been an absolute pleasure, it's great to see a project so welcoming of new contributions!

Reply:
If something is not supported, you can always do it manually (check and use `module.fail_json()`).

But if all combinations are ok, it's fine for me too. (Doing nothing by default is acceptable IMO.)

About the review process: let me mention that I'm not familiar with this module (or pkgng), so I can't provide a complete review on this one :) I hope that one of the maintainers can take a look at this one too...